### PR TITLE
Fold processing model into semantics section

### PR DESCRIPTION
**Pull request by glyn**
Fixes https://github.com/ietf-wg-jsonpath/draft-ietf-jsonpath-base/issues/199

### CODE REVIEW

File: draft-ietf-jsonpath-base.md [hunk 1]
```diff
@@ -457,56 +474,19 @@ nodelist, which are then concatenated into
...
-But a selector doesn't just act on a single node: a selector acts on
-each of the nodes in its input nodelist and concatenates the resultant nodelists
-to form the result nodelist of the selector.
-
-
-For each node in the list, the selector selects zero or more nodes,
-each of which is a descendant of the node or the node itself.
-
-For instance, with the argument `{"a":[{"b":0},{"b":1},{"c":2}]}`, the
+Let's walk through an example in detail. With the argument `{"a":[{"b":0},{"b":1},{"c":2}]}`, the
```
[the remark this answers is not in the document]
Comment (timbray) [reply draft-ietf-jsonpath-base.md:487]: Never mind, I found #199.  I'm OK with this PR.

[the remark this answers is not in the document]
Comment (glyn) [reply draft-ietf-jsonpath-base.md]: > Language such as "consider this example:" is a bit more typical for RFCs than "Let's walk".

Fixed, thanks.

File: draft-ietf-jsonpath-base.md [hunk 2]
```diff
@@ -422,25 +422,42 @@ type errors when pondering well-formedness and validity, while
...
+                               list-selector))
+~~~~
+
+The syntax and semantics of each selector is defined below.
 
-In this specification, the semantics of a JSONPath query are defined
-in terms of a *processing model*.  That model is not prescriptive of
-the internal workings of an implementation:  Implementations may wish
-(or need) to design a different process that yields results that are
-consistent with this model.
```
Comment (gregsdennis) [inline draft-ietf-jsonpath-base.md:431]: I think the "consistent with this model" sentence is fairly important to keep.  We're still defining a semantic "how it works."  It's important to say that implementations can do what they want so long as the output is what is expected.

Comment (glyn) [reply draft-ietf-jsonpath-base.md:431]: @gregsdennis Do you think "consistent with this model" needs adding, given the following opening sentence of this section?

> In this specification, the semantics of a JSONPath query define the
required results and do not prescribe the internal workings of an
implementation.

Comment (gregsdennis) [reply draft-ietf-jsonpath-base.md:431]: Ah. I think that's fine.

File: draft-ietf-jsonpath-base.md [hunk 3]
```diff
@@ -579,8 +559,8 @@ root-selector  = "$"
 #### Semantics
 {: unnumbered}
 
-The Argument — the root JSON value — becomes the root node, which is
-addressed by the root selector `$`.
+The root selector `$` selects the root node of the argument
+and produces a nodelist consisting of that root node.
```
Comment (sthagen) [inline draft-ietf-jsonpath-base.md:563]: Is the result of the root selector always a nodelist?

**Review by sthagen — APPROVED** [review]
Regardless of my question I like the change.

File: draft-ietf-jsonpath-base.md (hunk 3 above)
Comment (glyn) [reply draft-ietf-jsonpath-base.md:563]: Yes.

**Review by timbray — APPROVED** [review]

File: draft-ietf-jsonpath-base.md [hunk 4]
```diff
@@ -457,56 +474,19 @@ nodelist, which are then concatenated into
 the result of the selector.
 
 The processing within a selector may execute nested queries,
-which are in turn handled with the processing model defined here.
+which conform to the semantics defined here.
 Typically, the argument to that query will be the current node of the
 selector or a set of nodes subordinate to that current node.
 
-
```
[the remark this answers is not in the document]
Comment (glyn) [reply draft-ietf-jsonpath-base.md:464]: I don't think so. What do you think is missing?

[the remark this answers is not in the document]
Comment (cabo) [reply draft-ietf-jsonpath-base.md:464]: I don't have time at the moment, but I'm not sure where the processing model went.

[the remark this answers is not in the document]
Comment (glyn) [reply draft-ietf-jsonpath-base.md:464]: Ah, yes, I moved the processing model information into a top-level semantics section. The term "processing model" didn't seem to be adding any value. Please see the related issue https://github.com/ietf-wg-jsonpath/draft-ietf-jsonpath-base/issues/199 for the rationale for this change.

[the remark this answers is not in the document]
Comment (cabo) [reply draft-ietf-jsonpath-base.md:464]: Indeed, the term doesn't add anything.  It is a bit hard to follow text moving around in a github PR.

[the remark this answers is not in the document]
Comment (glyn) [reply draft-ietf-jsonpath-base.md:464]: Ok, if there's anything I can do to make the review easier, please let me know.

[the remark this answers is not in the document]
Comment (timbray) [reply draft-ietf-jsonpath-base.md:464]: FWIW I ended up looking in the branch at the .md file, which isn't beautifully formatted but good enough to be readable.
